roxie/ccd/ccdserver.cpp

```diff
@@ -7055,13 +7056,17 @@ class CRoxieServerDedupActivity : public CRoxieServerActivity
         numKept = 0;
         CRoxieServerActivity::start(parentExtractSize, parentExtract, paused);
         numToKeep = helper.numToKeep();
+        if (helper.keepBest())
+            compareBest = helper.queryCompareBest();
     }
 };
 
 class CRoxieServerDedupKeepLeftActivity : public CRoxieServerDedupActivity
 {
     IRangeCompare * stepCompare;
+    bool first;
```

Comment:
Again - use the keepRight version if best is specified and avoid the changes to this class.